x-pack/plugins/security_solution/public/detections/pages/alert_details/tabs/summary/index.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiFlexGroup } from '@elastic/eui';
+import type { TimelineEventsDetailsItem } from '@kbn/timelines-plugin/common';
+import { TimelineId } from '../../../../../../common/types';
+import { useDetailPanel } from '../../../../../timelines/components/side_panel/hooks/use_detail_panel';
+import { useGetUserCasesPermissions } from '../../../../../common/lib/kibana';
+import type { SelectedDataView } from '../../../../../common/store/sourcerer/model';
+import { SourcererScopeName } from '../../../../../common/store/sourcerer/model';
+import type { Ecs } from '../../../../../../common/ecs';
+import { AlertRendererPanel } from './alert_renderer_panel';
+import { RulePanel } from './rule_panel';
+import { CasesPanel, CasesPanelNoReadPermissions } from './cases_panel';
+import { HostPanel } from './host_panel';
+import { UserPanel } from './user_panel';
+import { SummaryColumn, SummaryRow } from './wrappers';
+
+export interface DetailsSummaryTabProps {
+  dataAsNestedObject: Ecs | null;
+  detailsData: TimelineEventsDetailsItem[];
+  sourcererDataView: SelectedDataView;
+}
+
+export const DetailsSummaryTab = React.memo(
+  ({ dataAsNestedObject, detailsData, sourcererDataView }: DetailsSummaryTabProps) => {
+    const eventId = dataAsNestedObject?._id as string;
```

Comment:
Is this cast safe when `dataAsNestedObject` is `null` per it's type definition:

```ts
dataAsNestedObject: Ecs | null;
```

Reply:
Technically `eventId === dataAsNestedObject._id` so it would have to exist for the page to even show. Rather than passing `eventId` down when the above equivalency exists, I just decided to use that instead....but `Ecs` strikes again and since the option can be undefined, I cast it here. Anyways, for simplicity, just went with passing `eventId`, but would really like to move away from `Ecs` in the future :joy: